ballerina/listener_oauth2_provider.bal

```diff
@@ -210,7 +210,22 @@ isolated function prepareIntrospectionResponse(json payload) returns Introspecti
                 introspectionResponse.tokenType = <string>payloadMap[key];
             }
             EXP => {
-                introspectionResponse.exp = <int>payloadMap[key];
+                anydata value = payloadMap[key];
+
+                if value is int {
+                    introspectionResponse.exp = value;
+                } else if value is string {
+                    int|error parsedInt = int:fromString(value);
+                    if parsedInt is int {
+                        introspectionResponse.exp = parsedInt;
+                    } else {
+                        introspectionResponse.exp = ();
+                        log:printError("Failed to parse string to integer for exp field", parsedInt);
+                    }
+                } else {
+                    introspectionResponse.exp = ();
+                    log:printError("Invalid type for exp field, expected int or string");
+                }
```

Comment:
```suggestion
                json value = payloadMap[key];
                if value is int {
                    introspectionResponse.exp = value;
                } else if value is string {
                    int|error parsedInt = int:fromString(value);
                    if parsedInt is int {
                        introspectionResponse.exp = parsedInt;
                    } else {
                        introspectionResponse.exp = ();
                        log:printError("Failed to parse string to integer for exp field", parsedInt);
                    }
                } else {
                    introspectionResponse.exp = ();
                    log:printError("Invalid type for exp field, expected int or string");
                }
```

Logic looks good. Lets move this logic into a separate function. 
```
isolated function parseExpClaim(json expClaim) returns int? {
     // logic
}
``

Reply:
Add a test for the `prepareIntrospectionResponse` covering the logic you have implemented.